baseplate.go

```diff
@@ -0,0 +1,245 @@
+package baseplate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/reddit/baseplate.go/batcherror"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/runtimebp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+// Config is a general purpose config for assembling a Baseplate server
+type Config struct {
+	// Addr is the local address to run your server on.
+	//
+	// It should be in the format "${IP}:${Port}", "localhost:${Port}",
+	// or simply ":${Port}".
+	Addr string
+
+	// Timeout is the socket connection timeout for Servers that
+	// support that.
+	Timeout time.Duration
+
+	// StopTimeout is the timeout for the Stop command for the service.
+	//
+	// If this is not set, then no timeout will be set on the Stop command.
+	StopTimeout time.Duration `yaml:"stopTimeout"`
```

Comment:
I think we should make the other YAML also explicit even though the mapping is correct implicitly. It communicates right away that this whole config is mapped to a file where just seeing this one `yaml` tag I'd thought only that one is mapped.